### PR TITLE
redux setup

### PR DESCRIPTION
**Pull request by ayu-ayushisood**

### CODE REVIEW

File: habit/package.json [hunk 1]
```diff
@@ -10,8 +10,12 @@
     "lint": "eslint ."
   },
   "dependencies": {
+    "@reduxjs/toolkit": "^1.4.0",
```
Comment (karanasthana) [inline habit/package.json:13]: What's this used for?

File: habit/package.json [hunk 2]
```diff
@@ -10,8 +10,12 @@
...
     "react": "16.13.1",
-    "react-native": "0.63.0"
+    "react-native": "0.63.0",
+    "redux": "^4.0.5",
+    "redux-logger": "^3.0.6",
```
Comment (karanasthana) [inline habit/package.json:17]: Do we really need a logger? 
What advantages does it give us?

File: habit/src/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React, {Component} from 'react';
+import * as configureStore from './store';
+import {Provider} from 'react-redux';
+
+export default class AppProvider extends Component {
+    render () {
+        return (
+          <Provider store={configureStore.STORE}>
+            <App />
```
Comment (karanasthana) [inline habit/src/index.js:9]: Why are we not importing App here?

File: habit/src/store/index.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import configureStore from './configure-store'
+
+export const STORE = configureStore();
```
Comment (karanasthana) [inline habit/src/store/index.js:3]: Can we do all this in one file only?

File: habit/src/store/configure-store.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { createStore, applyMiddleware, compose } from 'redux'
+import thunk from 'redux-thunk'
+import { logger } from 'redux-logger'
+import rootReducer from '../reducers/root-reducer'
+
+const middleware = [thunk];
```
Comment (karanasthana) [inline habit/src/store/configure-store.js:6]: We should re-think about logger.
If not using it, do something like
```
const store = createStore(rootreducer, applyMiddleware(thunk));

export default store;
```

File: habit/src/reducers/root-reducer.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import {combineReducers} from 'redux';
```
Comment (karanasthana) [inline habit/src/reducers/root-reducer.js:1]: We can shift this code into the index.js file of root-reducer

Comment (karanasthana) [reply habit/src/reducers/root-reducer.js:1]: Every folder has an index.js file, which is the first file called when importing something from there.
So, using the index.js file will reduce our imports from `import rootReducer from '../reducers/root-reducer';` to `import rootReducer from '../reducers/';`

File: habit/src/index.js (hunk 3 above)
Comment (ayu-ayushisood) [reply habit/src/index.js:9]: that's a mistake I have to import

File: habit/src/store/configure-store.js (hunk 5 above)
Comment (ayu-ayushisood) [reply habit/src/store/configure-store.js:6]: yeah, removing logger

File: habit/src/index.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React, {Component} from 'react';
```
Comment (karanasthana) [inline habit/src/index.js:1]: Should we have done this outside src?
Cannot think of a reason to do it inside src

Comment (karanasthana) [reply habit/src/index.js:1]: As in, the src/index.js

Comment (ayu-ayushisood) [reply habit/src/index.js:1]: why not inside? src should have the main index file.

File: habit/package.json (hunk 1 above)
Comment (ayu-ayushisood) [reply habit/package.json:13]: I'm not sure. From where I was reading it was written to use it for redux.

Comment (karanasthana) [reply habit/package.json:13]: Are we using it? @ayu-ayushisood 
If not, let's import only when we'll start using it

File: habit/src/index.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+import * as store from './store';
+import {Provider} from 'react-redux';
+import App from '../App';
+
+export default class AppProvider extends Component {
```
Comment (karanasthana) [inline habit/src/index.js:6]: This is not connected to the outside (main) index.js and app.js.
So, this code will not do anything in itself.

Need to fix the connection of the store to the App.js in the app

Comment (ayu-ayushisood) [reply habit/src/index.js:6]: there is a connection, the main app is imported and used as a component in this file.

**Pull request merged**